examples/bots/slack_bot.py

```diff
@@ -0,0 +1,153 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import asyncio
+import queue
+from typing import Optional
+
+from slack_bolt.context.async_context import AsyncBoltContext
+from slack_bolt.context.say.async_say import AsyncSay
+from slack_sdk.web.async_client import AsyncWebClient
+
+from camel.bots import SlackApp, SlackEventBody
+from examples.bots.agent import Agent
```

Comment:
for the example, I feel we need to make related code into one python file, if there's some abstraction need to be further added, it shouldn't be added into example folder

Reply:
`slack_bot` is an application of `slack_app`. Here we only introduce one usage, so we put it in the examples, and I think users may not necessarily use all platforms, so I think it makes sense to list them separately for them to choose.

Reply:
But for the `camel.bots` module, I have thought about abstracting them, but maybe after integrating more platforms and abstracting away the common process

Reply:
btw, what do you think about renaming `camel.bots` to `camel.adapters`, or something else? I think `bots` are also okay, but it always feels a bit strange that there are a bunch of platform apps in them. (although telegram is still a bot)

Reply:
let's keep the naming as `bots` as that's our target to integrate those apps